src/server/repositories/UrlRepository.ts

```diff
@@ -140,54 +132,14 @@ export class UrlRepository implements UrlRepositoryInterface {
     }
   }
 
-  public rawDirectorySearch: (
-    conditions: DirectoryQueryConditions,
-  ) => Promise<UrlDirectoryPaginated> = async (conditions) => {
-    const { query, order, limit, offset, state, isFile, isEmail } = conditions
-
-    const { tableName } = Url
-
-    const urlVector = urlSearchVector
-
-    const rankingAlgorithm = this.getRankingAlgorithm(order, tableName)
-
-    const urlsModel = await (isEmail
-      ? this.getRelevantUrlsFromEmail(
-          query,
-          rankingAlgorithm,
-          limit,
-          offset,
-          state,
-          isFile,
-        )
-      : this.getRelevantUrlsFromText(
-          urlVector,
-          rankingAlgorithm,
-          limit,
-          offset,
-          query,
-          state,
-          isFile,
-        ))
-
-    return urlsModel
-  }
-
-  private async getRelevantUrlsFromEmail(
-    query: string,
+  public async getRelevantUrlsFromEmail(
```

Comment:
Okay, this is a bad idea. you're allowing the arbitrary injection of SQL via `rankingAlgorithm`, `queryState` and `queryFile` through a public method. I know these are only accessed from a service, but I think that you have to trade-off between biz logic and security here.

Reply:
make sense, I have reverted the parsing back to urlrepository and made them private again